### ISSUE TITLE
File stops with Assertion failed: n == 4u

### ISSUE THREAD

**Issue by JasonGlazer** [NREL/EnergyPlus#6293]
Issue overview
--------------
A users file fails with message:

Assertion failed: n == 4u, file I:\epRepo\EnergyPlus-Develop\src\EnergyPlus\DataSurfaces.cc, line 517

Please ask Glazer for copy of file provided by user when working on issue #5672  

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by JasonGlazer** [comment]
This may be related to #5508

**Comment by mjwitte** [comment]
I though there was an existing issue about this, but I couldn't find it.

**Comment by jmarrec** [comment]
@JasonGlazer Do you still have said file? If so, can you put it on EnergyPlusDevSupport please?

**Comment by JasonGlazer** [comment]
@jmarrec I just emailed you the files

**Comment by jmarrec** [comment]
I cannot reproduce the assertion message with in 9.0.1. I updated the files (user's file are 8.5), fixed a couple issues (removed all `ComponentCost:LineItem`, and removed the `Unit Type` for all ScheduleTypeLimits (user's file are from an older openstudio version problably, which uses stuff like 'controlmode' or 'volumetricflowrate' which aren't in the E+ IDD).
I got errors in the eplusout.err about 6 degenerate surfaces, which I removed (as well as the surfaces that referenced them). I got an error about two surfaces that have mistmatched numbers of vertexes, removed these two. Then the file runs.

**Note:** I made a mistake initially, where I deleted a surface, but keep a surface that referenced it, and E+ crashed due to an assertion "contains". I think we should be able to handle that more gracefully and tell the user that we cannot find the surface that is referenced, I'll fix that in #7071.
